javascript/javascript1/week2/candyHelper.js

```diff
@@ -0,0 +1,47 @@
+const boughtCandyPrices = [];
+
+function addCandy(candyType, weight){
+    const pricePerGram = {
+        sweet: 0.5,
+        chocolate: 0.7,
+        toffee: 1.1,
+        chewingGum: 0.03,
+    }
+    if(candyType === 'sweet'){
+       boughtCandyPrices.push(pricePerGram.sweet * weight);
+    }else if(candyType === 'chocolate'){
+        boughtCandyPrices.push(pricePerGram.chocolate * weight);
+    }else if(candyType === 'toffee'){
+        boughtCandyPrices.push(pricePerGram.toffee * weight);
+    }else if(candyType === 'chewingGum'){
+        boughtCandyPrices.push(pricePerGram.chewingGum * weight);
+    }
+}
+addCandy('sweet', 20);
+addCandy('chocolate', 30);
+addCandy('toffee', 10);
+addCandy('chewingGum', 50);
+console.log(boughtCandyPrices);
+
+let boughtCandySum = 0;
+// for loops:
+// for (let i = 0; i < boughtCandyPrices.length; i++) {
+//     boughtCandySum += boughtCandyPrices[i];
+// }
+
+// while loops:
+let i = 0
+while (i < boughtCandyPrices.length) {
```

Comment:
`while` loops get the job done! 💪 But what would happen had you forgotten to increment `i` inside it? Generally, when going over arrays, I recommend using [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) as it is less error-prone. 😄 

If I were to call `canBuyMoreCandy` and was told that I can, and I then spend all my money. What would `canBuyMoreCandy` return afterwards? Can you see what might cause issues here? 🐛